### ISSUE TITLE
Best way to logout of OIDC when RP does not support RP-initiated logout?

### ISSUE THREAD

**Issue by omizrahi99** [panva/node-oidc-provider#835]
Hello,

I am currently dealing with an RP (Okta) that does not support RP-initiated logout. When logging out of Okta with this IdP configured, the browser redirects the user agent to the initial `/auth` endpoint, but node-oidc-provider automatically logs the user back in since the user's session cookies are still in the browser and database. What is the best way to log the user out of the IdP in this case?

Is there anyone way to clear the user's session when the `/auth` endpoint is hit by Okta clients, and before the check for the session cookies take place, so that the user won't be logged back in and they would be taken to the `/interaction/login` page instead?

Thank you.

**Comment by 6S** [comment]
Hello,

You can redirect the end-user to the authorization endpoint with the parameter `prompt=login` according to [OIDC Specification §3.1.2.1.](https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest) but keep in mind that the user will not will not be really be logged out in this case. It will only allow the user to change his account. Then you should name this option "Change account" on RP side.

**Comment by panva** [comment]
You can use any of the available extension points to make it so that login is required when coming from specific RPs, the interactions policy is one such point made for this.

**Comment by omizrahi99** [comment]
> Hello,
> 
> 
> 
> You can redirect the end-user to the authorization endpoint with the parameter `prompt=login` according to [OIDC Specification §3.1.2.1.](https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest) but keep in mind that the user will not will not be really be logged out in this case. It will only allow the user to change his account. Then you should name this option "Change account" on RP side.

How would this work? I don't have access to the RP and the way they send the initial request to the authorization endpoint.

**Comment by omizrahi99** [comment]
@panva Thank you for the response. I tried adding a check to the interactions policy under the login prompt, which checks if the client is Okta. However, now after successfully logging in, the browser is taken back to the login page instead of taken to Okta (the RP). This makes me think that the checks are being run right after the initial auth request, but also after logging in. 

Also, even if this solution worked, the session would still be present. Wouldn't it make more sense to be able to clear the session only once at the initial auth request coming from this specific client? How would this be achieved?

**Comment by omizrahi99** [comment]
Also, for reference, I am working off of example 03 from the step by step example.

**Comment by panva** [comment]
Of course the interaction policy runs on every pass. You can take account for that again in your interaction policy.

I'm not sure what your envisioned end state of the OP session is here.

**Comment by omizrahi99** [comment]
Thank you so much, I just realized the context in the check function gives me the request url, and I was able to use that to determine when to send the browser to the login page. It works now.

To clarify my thoughts on the session... To be truly logged out of the OP, the session needs to be cleared. In this case, forcing the user to be redirected back to the login page when the end user wants to sign out does not clear the OP session. I would appreciate it if you gave me your thoughts on how to clear the OP session when signing out of RPs that don't support RP-initiated logout.

Other than that, thank you so much for your help thus far!
